### PR TITLE
added functionality to add a proposal link to projects

### LINKED ISSUE #449: Add "Proposal" hotlink/button for every project
In the project description page, there should be a spot for a link to the proposal. Would be a convenient place to find all the proposals and quickly reference them in the future

### PR DESCRIPTION
**Pull request by jessamarie**
Added "proposalUrl" to the project model, an input box to the edit project component, a button to the project profile, and also a link on the project pages.

fixes #449

### CODE REVIEW

**Comment by vkr96** [comment]
Looks pretty good to me

File: client/app/projects/profile/profile.html [hunk 1]
```diff
@@ -28,6 +34,9 @@ <h1 >{{project.name}}</h1>
             <li><a href="{{repo}}" ng-repeat="repo in project.repositories">{{repo}}<span class="glyphicon glyphicon-menu-right"></span></a></li>
           </ul>
         </div>
+
```
Comment (agundy) [inline client/app/projects/profile/profile.html]: extra spacing?

File: client/app/projects/projects.html [hunk 2]
```diff
@@ -45,6 +45,7 @@ <h2 id="{{project.githubUsername}}-{{project.githubProjectName}}" class="project
             <li><a href="{{project.websiteUrl}}" target="_blank">Website</a></li>
             <li><a href="/projects/{{project.githubUsername}}/{{project.githubProjectName}}/blog">Blog</a></li>
             <li><a href="https://github.com/{{project.githubUsername}}/{{project.githubProjectName}}" target="_blank">Repository</a></li>
+            <li><a href="{{project.proposalUrl}}" target="_blank">Proposal</a></li>
```
Comment (agundy) [inline client/app/projects/projects.html]: Maybe don't show it if the field isn't set?

**Comment by kmcnellis** [comment]
I was thinking we should have projects fill out proposals right on observatory, in markdown.
Benefits: 

- We could pull in the description field from the profile (thereby having them update it each semester)
- Show a history of proposals from previous semesters, and how goals have changed
-- Could even have a checklist where people could check off goals through the semester
- Easy place for everyone to find said proposal
- Easy integration with auto-generating URPs
- Add an approval system for mentors to check proposals

Main downside: collaborative editing

**Comment by vkr96** [comment]
I think what Kiana is saying is a great feature, but I think this a good start to making propsal more accessible to students. Maybe make this a milestone for next semester, since this seems like a pretty big feature

**Comment by vkr96** [comment]
What do people think, merge this now or implement Kiana's features before doing so?

**Comment by agundy** [comment]
Merge for now after my suggested fix. Kiana's is more a future idea

**Comment by jessamarie** [comment]
@agundy made those changes. Is it okay to merge?

**Pull request merged**